frontend/app/api/farcaster/[userId]/route.ts

```diff
@@ -0,0 +1,23 @@
+import { publishNewUserKeysCast } from "@/lib/api/backend/farcaster";
+import { ERRORS } from "@/lib/errors";
+import prisma from "@/lib/prisma";
+import { SocialProfileType } from "@prisma/client";
+
+export async function POST(req: Request, { params }: { params: { userId: number } }) {
```

Comment:
I think that this endpoint shouldn't receive a userId. Otherwise I can post on behalf of other people.

I'd change to get the auth header from the request, look up the user based on that and then check if that user has a social profile or not.